spec/components/candidate_interface/degree_required_component_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::DegreeRequiredComponent, type: :component do
+  let(:application_form) { create(:application_form) }
+
+  let(:course_option1) { create(:course_option, course: create(:course, :open_on_apply, degree_grade: 'two_one')) }
+
+  let(:application_choice) do
+    build_stubbed(
+      :application_choice,
+      status: :unsubmitted,
+      course_option: course_option1,
+      application_form: application_form,
+    )
+  end
+
+  context 'application has no degree and course has requirement' do
+    it 'renders the degree row without guidance' do
+      result = render_inline(described_class.new(application_choice))
+      expect(result.text).to include('2:1 degree or higher (or equivalent)')
+    end
+  end
+
+  context 'application has other degree and course has requirement' do
+    it 'renders the degree row without guidance' do
+      create(
+        :degree_qualification,
+        qualification_type: 'Other Qual',
+        subject: 'Woof',
+        institution_name: 'University of Doge',
+        institution_country: 'GB',
+        grade: 'Lower second-class honours (2:2)',
+        predicted_grade: false,
+        start_year: '2005',
+        award_year: '2008',
+        application_form: application_form,
+      )
+
+      result = render_inline(described_class.new(application_choice))
+      expect(result.text).to include('2:1 degree or higher (or equivalent)')
+    end
+  end
+
+  context 'application has non_uk degree' do
+    it 'renders the degree row without guidance' do
+      create(
+        :degree_qualification,
```

Comment:
can probs just get away with creating a degree in most of these tests without all the other attrs right? E.g. for this one

create(
        :degree_qualification,
        grade: 'Lower second-class honours (2:2)',
        institution_country: 'Armenia',
        application_form: application_form,
      )

Reply:
Intended to but slipped my mind! Thanks for the 👀